src/nexus.cpp

```diff
@@ -49,26 +49,31 @@
  * This class is in charge of connecting various systems together
  * and forwarding signals appropriately to the right objects,
  * it is in charge of starting the GUI and the Core.
+ *
+ *
+ * @fn getProfile
+ * @brief Get current user profile.
+ * @return nullptr if not started, profile otherwise.
+ *
+ *
+ * @fn getDesktopGUI
+ * @brief Get desktop GUI widget.
+ * @return nullptr if not started, desktop widget otherwise.
  */
 
 Q_DECLARE_OPAQUE_POINTER(ToxAV*)
 
-static Nexus* nexus{nullptr};
-
-Nexus::Nexus(QObject *parent) :
-    QObject(parent),
-    profile{nullptr},
+Nexus::Nexus() :
+    QObject(),
+    mProfile{nullptr},
     widget{nullptr},
     loginScreen{nullptr}
 {
 }
 
 Nexus::~Nexus()
 {
-    delete widget;
-    delete loginScreen;
-    delete profile;
```

Comment:
1. Have a look [here](https://github.com/qTox/qTox/pull/4001/files#diff-39d71983c1fc16a6b3191ee87174baaeR96) and [here](https://github.com/qTox/qTox/pull/4001/files#diff-1e0d837757bc75752ac84d02d7e8ed44R40) -> double free -> crash
2. Widget and LoginScreen instances are not "owned" by the Nexus instance, but rather (mis-)used for page switching. So one of both members is always a `nullptr`. This is a classic use case for `QMainWindow::setCentralWidget`. Probably too much for the moment… :)